packages/clarity-js/src/layout/region.ts

```diff
@@ -1,108 +1,77 @@
-    }
-
-    if (changed) {
-        if (updateMap.indexOf(id) === -1) { updateMap.push(id); }
-        let value = dom.getValue(id);
-        let r = value ? dom.getRegion(value.region) : null;
-        bm[id] = { id, box, region: r };
-    }
+export function exists(node: Node): boolean {
+    return regionMap.has(node);
 }
 
 export function track(id: number): void {
-    if (updateMap.indexOf(id) === -1) {
-        updateMap.push(id);
+    // Do not send another entry if we are already sending it
+    let node = dom.getNode(id);
+    if (updates.indexOf(id) <= 0 && regionMap.has(node)) {
+        data.push(id in regions ? regions[id] : { id, visible: BooleanFlag.False, region: regionMap.get(node) });
     }
 }
 
-export function layout(element: Element, viewport: Box = null): Box {
-    let box: Box = null;
-    if (typeof element.getBoundingClientRect === "function") {
-        // getBoundingClientRect returns rectangle relative positioning to viewport
-        let rect = element.getBoundingClientRect();
-
-        if (rect && rect.width > 0 && rect.height > 0) {
-            viewport = viewport || getViewport();
-            let visible =
-                // Top of rectangle is lesser than the bottom of the viewport
-                rect.top <= viewport.h &&
-                // Bottom of rectangle is greater than the top of the viewport
-                rect.top + rect.height >= 0 &&
-                // Left of rectangle is lesser than the right of the viewport
-                rect.left <= viewport.w &&
-                // Right of rectangle is greater than the left of the viewport
-                rect.left + rect.width >= 0 &&
-                // The current page is visible
-                ("visibilityState" in document ? document.visibilityState === "visible" : true);
+export function compute(): void {
+    // Schedule encode only when we have at least one valid data entry
+    if (data.length > 0) { encode(Event.Region); }
+}
 
-            // Add viewport's scroll position to rectangle to get position relative to document origin
-            // Also: using Math.floor() instead of Math.round() because in Edge,
-            // getBoundingClientRect returns partial pixel values (e.g. 162.5px) and Chrome already
-            // floors the value (e.g. 162px). This keeps consistent behavior across browsers.
-            box = {
-                x: Math.floor(rect.left + viewport.x),
-                y: Math.floor(rect.top + viewport.y),
-                w: Math.floor(rect.width),
-                h: Math.floor(rect.height),
-                v: visible ? BooleanFlag.True : BooleanFlag.False
-            };
+function handler(entries: IntersectionObserverEntry[]): void {
+    for (let entry of entries) {
+        let target = entry.target;
+        let id = target ? dom.getId(target) : null;
+        if (id && regionMap.has(target)) {
+            let visible = entry.isIntersecting ? BooleanFlag.True : BooleanFlag.False;
```

Comment:
Have you thought about sending interactoinRatio and/or interactionRect too? You probably did but would be interested in the thinking.

Reply:
Yes but intentionally favoring simplicity here to keep script size small. We can always extend when we have a use case.